src/gpu/intel/ocl/reusable_softmax.cl

```diff
@@ -72,3 +74,140 @@ reusable_softmax_fwd_generic(__global SRC_DATA_T *src, __global DST_DATA_T *dst,
         dst[c - begin] = TO_DST(unscaled * scale);
     }
 }
+
+#endif
+
+#ifdef USE_VECTORIZED_KERNEL
+
+__attribute__((reqd_work_group_size(SUBGROUP_SIZE, 1, 1)))
+__attribute__((intel_reqd_sub_group_size(SUBGROUP_SIZE))) __kernel void
+reusable_softmax_fwd_generic(__global DATA_T *src, __global DST_DATA_T *dst,
+        __global float *src_scale, __global float *dst_scale,
+        dim_t SOFTMAX_AXIS_SIZE, dim_t softmax_axis_stride,
+        dim_t softmax_axis_chunk_size, dispatch_gws_rt_params_t gws_params) {
+
+    float scale = 1.0f;
+
+    const off_t linear_thread_id = get_global_id(0);
+    const off_t data_off
+            = (linear_thread_id / SUBGROUP_SIZE) * SOFTMAX_AXIS_SIZE;
+    global DATA_T *src_backup = src;
+    src += data_off;
+
+    VECT_FLOAT_T dk;
```

Comment:
You should consider using an array type here. In my experience, the vector types just cause problems as they lack the same size flexibility of array types, while at the same time, they don't provide any performance benefit.

Reply:
Moving to array vs vector will be handled in an additional PR. See comment https://github.com/uxlfoundation/oneDNN/pull/2525#discussion_r2083530745 regarding `block_load`. Will make the vector to array switch once the issue with `block_load` is resolved.